### PR TITLE
206-fix-convert-to-xliff

### PR DESCRIPTION
**Pull request by klavinsjanis**
To address randomly failing integration tests in:
203-conflict-when-multiple-original-translations

### CODE REVIEW

File: pkg/convert/xliff12.go [hunk 1]
```diff
@@ -96,25 +96,37 @@ func ToXliff12(translation model.Translation) ([]byte, error) {
...
 				TransUnits: make([]transUnit, 0, len(translation.Messages)),
 			},
 		},
 	}
 
+	if translation.Original {
+		xlf.File.SourceLanguage = translation.Language
+	} else {
+		xlf.File.TargetLanguage = translation.Language
+	}
```
Comment (janishorsts) [inline pkg/convert/xliff12.go:109]: When flow is without an else block, the compiled binary code is identical to your flow `if ... else`. I like this better. Your call.

```suggestion
	xlf.File.TargetLanguage = translation.Language
	if translation.Original {
		xlf.File.SourceLanguage = translation.Language
	}
```

Comment (klavinsjanis) [reply pkg/convert/xliff12.go:109]: I am confused, how is that identical?

File: pkg/convert/xliff2.go [hunk 2]
```diff
@@ -104,16 +104,27 @@ func ToXliff2(translation model.Translation) ([]byte, error) {
 		},
 	}
 
+	if translation.Original {
+		xlf.SrcLang = translation.Language
+	} else {
+		xlf.TrgLang = translation.Language
```
Comment (janishorsts) [inline pkg/convert/xliff2.go:110]: Does xliff12 have only TrgLang or both when it is translated?

* when original, only SrcLang is set
* when not original, SrcLang and TgtLang are set

Comment (klavinsjanis) [reply pkg/convert/xliff2.go:110]: This is just a minor fix, addressing the failing integration tests. The implementation is incomplete for convert to XLIFF. In the current state it's not accurate/incomplete. I have added a ticket:
https://github.com/orgs/translate-agent/projects/1?pane=issue&itemId=45498477

**Review by janishorsts — APPROVED** [review]

**Pull request merged**